tests/unit/test_diffsync.py

```diff
@@ -468,6 +468,26 @@ def callback(stage, current, total):
     assert last_value == {"current": expected, "total": expected}
 
 
+def test_diffsync_sync_to_w_diff(backend_a, backend_b):
+    diff = backend_b.diff_to(backend_a)
+    assert diff.has_diffs()
+    # Mock diff_from to make sure it's not called when passing in an existing diff
+    backend_b.diff_from = mock.Mock()
```

Comment:
For extra certainty, maybe mock both `diff_from` and `diff_to` on both `backend_a` and `backend_b` and assert that none of the four get called?